### ISSUE TITLE
ETHPeerPool._periodically_report_metrics crashes (with a PeerConnectionLost) and takes trinity with it

### ISSUE THREAD

**Issue by gsalgado** [ethereum/trinity#1732]
### What is wrong?

Got this crash today.

```
   DEBUG  2020-05-19 15:59:46,855               Manager  <Manager[ETHPeerPool] flags=SRcfe>: task _periodically_report_metrics[daemon=True] exited with error: Cannot look up subprotocol when connection is closing
Traceback (most recent call last):
  File "/home/salgado/src/async-service/async_service/base.py", line 300, in _run_and_manage_task
    await task.run()
  File "/home/salgado/src/async-service/async_service/asyncio.py", line 35, in run
    await self._async_fn(*self._async_fn_args)
  File "/home/salgado/src/trinity/p2p/peer_pool.py", line 505, in _periodically_report_metrics
    self._peer_reporter_registry.trigger_peer_reports()
  File "/home/salgado/src/trinity/p2p/metrics.py", line 54, in trigger_peer_reports
    self.make_periodic_update(peer, peer_id)
  File "/home/salgado/src/trinity/trinity/protocol/common/peer.py", line 171, in make_periodic_update
    head_gauge.set_value(peer.head_info.head_number)
  File "/home/salgado/virtualenvs/trinity-asyncio/lib/python3.7/site-packages/cached_property.py", line 35, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/home/salgado/src/trinity/trinity/protocol/common/peer.py", line 105, in head_info
    return self.connection.get_logic(HeadInfo.name, HeadInfo)
  File "/home/salgado/src/trinity/p2p/connection.py", line 265, in get_logic
    if not self.has_logic(name):
  File "/home/salgado/src/trinity/p2p/connection.py", line 261, in has_logic
    raise PeerConnectionLost("Cannot look up subprotocol when connection is closing")
p2p.exceptions.PeerConnectionLost: Cannot look up subprotocol when connection is closing
   DEBUG  2020-05-19 15:59:46,866               Manager  <Manager[ETHPeerPool] flags=SRCfE>: _handle_cancelled triggering task cancellation
```

After that, `EthPeerPool` terminates and takes trinity with it

### How can it be fixed

Guess we need to catch `PeerConnectionLost` in yet another place, @cburgdorf?

**Comment by gsalgado** [comment]
Hmm, and I just realized I did not explicitly enable the metrics component, so that should not even be running in the first place, no?

**Comment by cburgdorf** [comment]
>Hmm, and I just realized I did not explicitly enable the metrics component, so that should not even be running in the first place, no?

True. Generally, the metrics code runs with noop gauges, counters etc when `--metrics` isn't passed so that we do not have to sprinkle `if metrics_not_enabled:`all over the place but I agree that it wouldn't hurt to here if we just decided to not run the periodic update at all if the metrics system isn't enabled.

Pinging @njgheorghita here who worked on this and might want to continue in that area. Let me know if you are busy and I handle the fix.

**Comment by njgheorghita** [comment]
@cburgdorf My plate's a little full today - but I can probably dig into this tomorrow. Though, if it's not too much of a burden, it might be more efficient for you to knock this one out :wink:

**Comment by cburgdorf** [comment]
@njgheorghita It's going to be the first thing I look into this morning.

**Comment by cburgdorf** [comment]
This was fixed with #1736
